prescient/engine/egret/egret_plugin.py

```diff
@@ -552,15 +557,11 @@ def solve_deterministic_day_ahead_pricing_problem(solver, ruc_results, options,
                 surplus_multiple_t = 1.
             else:
                 ## scale the reserves from the RUC down by the same fraction
-                ## so that they exactly meed the needed reserves
+                ## so that they exactly meet the needed reserves
                 surplus_multiple_t = reserve_requirement_t/reserve_provided_t
-            for g, reserve_vals in g_reserve_values.items():
-                thermal_reserve_cleared_DA[g,t] = reserve_vals[t]*surplus_multiple_t
-    else:
-        day_ahead_reserve_prices = { t : 0. for t,_ in enumerate(ruc_results.data['system']['time_keys']) }
-        thermal_reserve_cleared_DA = { (g,t) : 0. \
-                for t,_ in enumerate(ruc_results.data['system']['time_keys']) \
-                for g,_ in ruc_results.elements(element_type='generator', generator_type='thermal') }
+
+            thermal_reserve_cleared_DA[reserve_prod] = {(g,t): reserve_vals[t]*surplus_multiple_t
+                                                        for g, reserve_vals in g_reserve_values.items()}
```

Comment:
This is more a note to myself -- we probably need to be doing something else here because the existence of reserves-by-zone alongside system-level reserves makes the original logic that was here invalid.

Reply:
It isn't possible to 100% resolve this with the current Egret properties - if a generator takes part in multiple reserve products of the same type (such as `spinning_reserve`), Egret's corresponding attribute (`spinning_reserve_provided`) is ambiguous. 

The new code does distinguish between reserve products (it used to use `rg` for all reserve products, which was just wrong), and does ensure each generator participates in the reserve product before storing its day-ahead clearance.